cabal-install/src/Distribution/Client/CmdInstall.hs

```diff
@@ -742,7 +744,13 @@ getClientInstallFlags :: Verbosity -> GlobalFlags -> ClientInstallFlags -> IO Cl
 getClientInstallFlags verbosity globalFlags existingClientInstallFlags = do
   let configFileFlag = globalConfigFile globalFlags
   savedConfig <- loadConfig verbosity configFileFlag
-  pure $ savedClientInstallFlags savedConfig `mappend` existingClientInstallFlags
+  isSandbox <- Sandbox.isSandbox Nothing
+  let merged = savedClientInstallFlags savedConfig `mappend` existingClientInstallFlags
+  if isSandbox
+  then do
+    cwd <- getCurrentDirectory
+    pure $ merged { cinstEnvironmentPath = Flag cwd }
+  else pure $ merged
```

Comment:
Here we check during `cabal install` whether we're in a sanbdox and if so pass the `--package-env=.` flag, creating a ghc environment file.

It seems the env file already acts as a "world file"... it will (except for `base`) only contain the targets we specified via `cabal install --lib`. So I'm not sure it's necessary to play tricks with recreation of the env file to allow new resolutions.

Instead, I think it is reasonable to have users specify multiple targets to avoid resolution issues, such as `cabal install --lib text attoparsec`.